src/ai/backend/web/proxy.py

```diff
@@ -127,22 +127,33 @@ async def close_upstream(self) -> None:
             await self.up_conn.close()
 
 
-async def decrypt_payload(request: web.Request) -> str:
-    if not request.content:
-        return ""
-    config = request.app["config"]
-    scheme = config["service"]["force_endpoint_protocol"]
-    if scheme is None:
-        scheme = request.scheme
-    api_endpoint = f"{scheme}://{request.host}"
-    payload = await request.text()
-    iv, real_payload = payload.split(":")  # Extract initial vector and actual payload
-    key = (base64.b64encode(api_endpoint.encode("ascii")).decode() + iv + iv)[0:32]
-    crypt = AES.new(bytes(key, encoding="utf8"), AES.MODE_CBC, bytes(iv, encoding="utf8"))
-    b64p = base64.b64decode(real_payload)
-    dec = unpad(crypt.decrypt(bytes(b64p)), 16)
-    result = dec.decode("UTF-8")
-    return result
+@web.middleware
+async def decrypt_payload(request: web.Request, handler) -> web.StreamResponse:
+    request_headers = extra_config_headers.check(request.headers)
+    secure_context = request_headers.get("X-BackendAI-Encoded", None)
+    if secure_context:
+        if not request.content:
+            request.app["config"]["payload"] = ""
+            return await handler(request)
+        config = request.app["config"]
+        scheme = config["service"]["force_endpoint_protocol"]
+        if scheme is None:
+            scheme = request.scheme
+        api_endpoint = f"{scheme}://{request.host}"
+        payload = await request.text()
+        iv, real_payload = payload.split(":")  # Extract initial vector and actual payload
+        key = (base64.b64encode(api_endpoint.encode("ascii")).decode() + iv + iv)[0:32]
+        crypt = AES.new(bytes(key, encoding="utf8"), AES.MODE_CBC, bytes(iv, encoding="utf8"))
+        b64p = base64.b64decode(real_payload)
+        dec = unpad(crypt.decrypt(bytes(b64p)), 16)
+        result = dec.decode("UTF-8")
+        if not result:
+            request.app["config"]["payload"] = {}
```

Comment:
`request.app` is not suitable for storing request-specific data; it's designed for storing global variables (ref: https://docs.aiohttp.org/en/stable/web_advanced.html#application-s-config). Request data can be stored directly in `request`.
```suggestion
            request["payload"] = {}
```